src/plugins/home/server/services/sample_data/data_sets/util.ts

```diff
@@ -74,6 +78,30 @@ export const getSavedObjectsWithDataSource = (
         ) {
           saveObject.attributes.title = saveObject.attributes.title + `_${dataSourceTitle}`;
         }
+
+        if (saveObject.type === 'visualization') {
+          const vegaSpec = extractVegaSpecFromSavedObject(saveObject);
+
+          if (!!vegaSpec) {
+            const updatedVegaSpec = updateDataSourceNameInVegaSpec({
+              spec: vegaSpec,
+              newDataSourceName: dataSourceTitle,
+              spacing: 1,
```

Comment:
The reason `spacing` is 1 is because there is one sample data visualization (sankey visualization) that would cause a URL too long error. This setting limits the spacing so the visualization will still be rendered/accessed.

Reply:
Can we add those explanation as comments?